### PR TITLE
Added logic to clear modules from memory so they can be reloaded

### PR DESCRIPTION
**Pull request by bzwheeler**

### CODE REVIEW

File: src/pypyjs.js [hunk 1]
```diff
@@ -472,18 +477,24 @@ pypyjs.prototype.addModule = function addModule(name, source) {
...
 // This is the basic way to push code into the pypyjs VM.
 // Calling code should not use it directly; rather we use it
 // as a primitive to build up a nicer execution API.
 //
-pypyjs.prototype._execute_source = function _execute_source(code) {
+pypyjs.prototype._execute_source = function _execute_source(code, preCode) {
   console.log('executing: ' + code);
   const Module = this._module;
   let code_ptr;
+  if (!preCode) {
```
Comment (albertjan) [inline src/pypyjs.js]: I don't have the linter installed yet but I'm pretty sure it won't like reassignment of function args :smile: 

`const _preCode = preCode ? preCode : '';`

File: src/pypyjs.js [hunk 2]
```diff
@@ -566,10 +582,27 @@ pypyjs.prototype.exec = function exec(code) {
...
+    // sys.modules which clears them from memory and allows them to be reloaded
+    // from the emscripten file system
+    if (Object.keys(this._modulesToReset).length) {
+      // construct python code to remove module from sys.modules:
+      // ```python
+      //   import sys
+      //   if 'foo' in sys.modules: del(sys.modules['foo'])
+      // ```
+      preCode = 'try:\n  import sys\n';
+      for (let module of Object.keys(this._modulesToReset)) {
```
Comment (albertjan) [inline src/pypyjs.js]: Without trying to sounds like an ass :smile: could you write this as a `map` function?

**Comment by albertjan** [comment]
@bzwheeler fixed my own comments. Could you check if it still works?

**Comment by bzwheeler** [comment]
Almost worked, had some extra indentation :-)

**Comment by albertjan** [comment]
Oh sorry I hate when things almost work ...

**Pull request merged**